contracts/upgradeability/ClassicEternalStorageProxy.sol

```diff
@@ -4,32 +4,28 @@ import "./EternalStorage.sol";
 import "./OwnedUpgradeabilityProxy.sol";
 
 
-contract ClassicEternalStorageProxy is OwnedUpgradeabilityProxy, EternalStorage {
+contract ClassicEternalStorageProxy is EternalStorage, OwnedUpgradeabilityProxy {
```

Comment:
I have to re-order the inheritance of the contracts because we weren't able to read the correct EternalStorage values from the proxy contract. For example on this line:
```
uint256 ret = uintStorage[keccak256("dataSizes", sig)];
``` 
The order is important because of how slot of memory are assigned. Before, proxy storage was assigning:
[0] owner of proxy
[1] version
[2] implementation
[3...] all eternal storage mappings

While on implementation contract it was assigning all eternal storage mappings on firsts slots. So when methods calls were delegated to implementation contracts there was no issues on reading and writing the values, but when trying to read o write values from proxy contracts it will be reading/writing on differents slots.

Here is an example on how EternalStorageProxy is defined on zeppelin repo: https://github.com/zeppelinos/labs/blob/master/upgradeability_using_eternal_storage/contracts/EternalStorageProxy.sol

Here is detailed explanation of this https://github.com/zeppelinos/labs/tree/master/upgradeability_using_eternal_storage

Does this makes sense to you?